### PR TITLE
GIX-1978: New MyTokens route

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Implement a new page to show all the tokens of the user.

In this PR, setup the route for `/my-tokens`.

# Changes

* New page and layout under `my-tokens` directory in `routes`.
* New page component MyTokens to be developed.
* Check login in the MyTokens route.
* Setup new page SignInMyTokens.svelte.
* Add `MyTokens` to `AppPath`.

# Tests

* New PO for MyTokens route.
* New PO for MyTokens page.
* New PO for SignInMyTokens page.
* Test that MyTokens page renders different depending on auth.

# Todos

- [ ] Add entry to changelog (if necessary).
Not yet, I prefer to add it when the functionality is ready under a feature flag.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@dskloetd @mstrasinskis please review

**Comment by peterpeterparker** [comment]
Why call it `my-tokens` and not just `tokens`?

File: frontend/src/tests/page-objects/MyTokensPage.page-object.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export class MyTokensPagePo extends BasePageObject {
+  private static readonly TID = "my-tokens-page-component";
+
+  static under(element: PageObjectElement): MyTokensPagePo {
+    return new MyTokensPagePo(element.byTestId(MyTokensPagePo.TID));
+  }
+
+  hasLoginPage(): Promise<boolean> {
+    return SignInMyTokensPagePo.under(this.root).isPresent();
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/MyTokensPage.page-object.ts:13]: The component with `data-tid=sign-in-my-tokens-page-component` is not a child of the component with `data-tid=my-tokens-page-component` so this should not be possible.

File: frontend/src/tests/page-objects/MyTokensPage.page-object.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  hasMyTokensPage(): Promise<boolean> {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/MyTokensPage.page-object.ts:16]: Can the caller of this just call `.isPresent()` instead?

File: frontend/src/tests/routes/app/my-tokens/page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { render } from "@testing-library/svelte";
+
+describe("MyTokens route", () => {
+  const renderPage = () => {
+    const { container } = render(MyTokensRoute);
+
+    return MyTokensRoutePo.under(new JestPageObjectElement(container));
+  };
+
+  describe("when logged in", () => {
```
Comment (dskloetd) [inline frontend/src/tests/routes/app/my-tokens/page.spec.ts:14]: Test with the feature flag (explicitly) enabled *and* disabled?

**Review by dskloetd — COMMENTED** [review]
I agree with David that it's weird to have "my" everywhere. I thought I already mentioned this and you didn't seem to disagree?

**Comment by lmuntaner** [comment]
> Why call the route `/my-tokens` and not just `/tokens`?

Done!

**Comment by lmuntaner** [comment]
> I agree with David that it's weird to have "my" everywhere. I thought I already mentioned this and you didn't seem to disagree?

I don't remember agreeeing exactly on something else. I went with `/tokens` and Tokens name

File: frontend/src/tests/routes/app/my-tokens/page.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/routes/app/my-tokens/page.spec.ts]: Good point, done!

File: frontend/src/tests/page-objects/MyTokensPage.page-object.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/MyTokensPage.page-object.ts]: Yes, this was residual from copy pasting, thanks!

File: frontend/src/tests/page-objects/MyTokensPage.page-object.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/MyTokensPage.page-object.ts]: Yes, this was residual from copy pasting, thanks!

**Comment by lmuntaner** [comment]
@peterpeterparker @dskloetd ready for another review

File: frontend/src/tests/routes/app/tokens/page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      it("should render sign-in if not logged in", async () => {
+        const po = renderPage();
+
+        expect(await po.hasLoginPage()).toBe(true);
+        expect(await po.hasTokensPage()).toBe(false);
+      });
+    });
+  });
+
+  describe("when feature flag enabled", () => {
```
Comment (dskloetd) [inline frontend/src/tests/routes/app/tokens/page.spec.ts:51]: enabled => disabled

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/tests/routes/app/tokens/page.spec.ts]: thanks!

**Pull request merged**